org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/AST.java

```diff
@@ -1193,7 +1204,7 @@ private AST(int level, boolean previewEnabled) {
 						false/*isPreviewEnabled*/);
 				break;
 			default:
-				if (level < JLS2_INTERNAL && level > JLS_Latest) {
+				if (!ALL_VERSIONS.contains(level)) {
```

Comment:
Should it be `SUPPORTED_VERSIONS` ?

Reply:
Using SUPPORTED_VERSIONS should happen in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3533 together with switching the mapping to prevent inconsistencies.